src/widgets/buttons/SvgButton.cpp

```diff
@@ -65,7 +71,16 @@ void SvgButton::paintContent(QPainter &painter)
     QSize actualPadding = this->scale() * this->padding_;
     QPoint topLeft{actualPadding.width(), actualPadding.height()};
     QSize contentSize = this->size() - 2 * actualPadding;
-    this->svg_->render(&painter, {topLeft, contentSize});
+    auto bounds = QRectF{topLeft, contentSize};
+    this->svg_->render(&painter, bounds);
+
+    if (this->color_.has_value())
+    {
+        painter.save();
+        painter.setCompositionMode(QPainter::CompositionMode_SourceIn);
+        painter.fillRect(bounds, *this->color_);
+        painter.restore();
```

Comment:
There should be a comment here explaining what this does, because it's not entirely obvious.

Reply:
added a comment about it in [`59135a1` (#6268)](https://github.com/Chatterino/chatterino2/pull/6268/commits/59135a15c21379f91f84bd4b2198df06c23eee75) and linked to https://doc.qt.io/qt-6/qpainter.html#CompositionMode-enum which explains it in more detail